test/beacon/Voucher.test.ts

```diff
@@ -405,5 +409,151 @@ describe('Voucher', function () {
                 ),
             ).to.be.revertedWith('IexecPocoMock: Failed to sponsorMatchOrders');
         });
+        describe('Match orders boost', async function () {
+            it('Should match orders boost with full sponsored amount', async () => {
+                for (const asset of [app, dataset, workerpool]) {
+                    await voucherHubWithAssetEligibilityManagerSigner
+                        .addEligibleAsset(voucherType, asset)
+                        .then((x) => x.wait());
+                }
+                const voucherInitialCreditBalance = await voucher.getBalance();
+                const voucherInitialSrlcBalance = await getVoucherBalanceOnIexecPoco();
+                const requesterInitialSrlcBalance = await getRequesterBalanceOnIexecPoco();
+
+                expect(
+                    await voucherWithOwnerSigner.matchOrdersBoost.staticCall(
+                        appOrder,
+                        datasetOrder,
+                        workerpoolOrder,
+                        requestOrder,
+                    ),
+                ).to.be.equal(dealId);
+                await expect(
+                    voucherWithOwnerSigner.matchOrdersBoost(
+                        appOrder,
+                        datasetOrder,
+                        workerpoolOrder,
+                        requestOrder,
+                    ),
+                )
+                    .to.emit(voucher, 'OrdersBoostMatchedWithVoucher')
```

Comment:
Check `VoucherDebited` and `Transfert` events.

Reply:
Those events are related to VocherHub / SRLC contracts, out of scope here (?)

Reply:
Ok for `VoucherDebited` but we have `IERC20(iexecPoco).transferFrom()` inside the function.

Reply:
Done [here](https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/pull/16/commits/c8ea51e8e3ef16cb9c213830cda75d54d38b61b4)